### ISSUE TITLE
[BUG] Documentation and informative logs: eventhubs and servicebus namespace property needs clarification

### ISSUE THREAD

**Issue by lmolkova** [Azure/azure-sdk-for-java#27053]
```
spring.cloud.azure.eventhubs.namespace |  String | The namespace of an event hub.
spring.cloud.azure.servicebus.namespace | String | Service Bus Namespace value.
```
1. `namespace` is `<namespace>.servicebus.windows.net` and not FQDN - it's worth clarifying in property descriptions
2. nit: what's the benefit of separating domain-name and namespace? why not just have fqdn property?
3. There is no validation of this field, i.e. it's hard to understand what went wrong without reading logs and noticing wrong FQDN (e.g. below is the only indication i get that something went wrong when I configure FQDN instead of namespace in [this sample](https://github.com/Azure-Samples/azure-spring-boot-samples/blob/spring-cloud-azure_4.0.0-beta.4/eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration)). Is it possible to validate it and throw a descriptive exception? 

```
2022-02-11 11:57:55.418  WARN 1784 --- [ctor-executor-3] c.a.c.a.i.handler.ConnectionHandler      : onTransportError hostname[azmondemo.servicebus.windows.net.servicebus.windows.net], connectionId[MF_6ed7da_1644609470211], error[null]
2022-02-11 11:57:39.896 ERROR 1784 --- [ctor-executor-1] reactor.core.publisher.Operators         : Operator called default onErrorDropped
reactor.core.Exceptions$ErrorCallbackNotImplemented: com.azure.core.amqp.exception.AmqpException: errorContext[NAMESPACE: azmondemo.servicebus.windows.net.servicebus.windows.net. ERROR CONTEXT: N/A]
Caused by: com.azure.core.amqp.exception.AmqpException: errorContext[NAMESPACE: azmondemo.servicebus.windows.net.servicebus.windows.net. ERROR CONTEXT: N/A]
	at com.azure.core.amqp.implementation.ExceptionUtil.toException(ExceptionUtil.java:85) ~[azure-core-amqp-2.3.5.jar:2.3.5]
	at com.azure.core.amqp.implementation.handler.ConnectionHandler.notifyErrorContext(ConnectionHandler.java:325) ~[azure-core-amqp-2.3.5.jar:2.3.5]
	at com.azure.core.amqp.implementation.handler.ConnectionHandler.onTransportError(ConnectionHandler.java:228) ~[azure-core-amqp-2.3.5.jar:2.3.5]
	at org.apache.qpid.proton.engine.BaseHandler.handle(BaseHandler.java:191) ~[proton-j-0.33.8.jar:na]
	at org.apache.qpid.proton.engine.impl.EventImpl.dispatch(EventImpl.java:108) ~[proton-j-0.33.8.jar:na]
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.dispatch(ReactorImpl.java:324) ~[proton-j-0.33.8.jar:na]
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.process(ReactorImpl.java:291) ~[proton-j-0.33.8.jar:na]
	at com.azure.core.amqp.implementation.ReactorExecutor.run(ReactorExecutor.java:92) ~[azure-core-amqp-2.3.5.jar:2.3.5]
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68) ~[reactor-core-3.4.9.jar:3.4.9]
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28) ~[reactor-core-3.4.9.jar:3.4.9]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]```

**Comment by saragluna** [comment]
Thanks for creating this issue, @yiliuTo please take a look at this issue.

**Comment by yiliuTo** [comment]
@lmolkova thanks for reporting this. 
1. We will clarify the meaning of namespace property in docs to help users distinguish it from FQDN.
2. The reason for having separated properties instead of FQDN directly is that we also the namespace property for the aim of [resource provisioning](https://github.com/Azure/azure-sdk-for-java/blob/3a1c944c376169588412248fb8667307390faa1e/sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/config/ServiceBusBinderConfiguration.java#L72) of Binder libraries. So we choose to use two individual properties.
3. About the validation, we will consider to add it and throw meaningful logs to help users understand.

**Comment by lmolkova** [comment]
@yiliuTo thanks, profile.tenant-id was the issue!

**Comment by saragluna** [comment]
Closed in #27450.
